sqldelight-compiler/src/test/kotlin/com/squareup/sqldelight/core/BindArgsTest.kt

```diff
@@ -165,6 +166,54 @@ class BindArgsTest {
     }
   }
 
+  @Test fun `bind args for upsert do update statements inherit type from column`() {
+    val file = FixtureCompiler.parseSql("""
+      |CREATE TABLE data (
+      |  id INTEGER PRIMARY KEY NOT NULL,
+      |  list INTEGER AS kotlin.collections.List NOT NULL
+      |);
+      |
+      |someUpsert:
+      |INSERT INTO data(
+      |  id, list
+      |)
+      |VALUES(
+      |  :id, :list
+      |)
+      |ON CONFLICT(
+      |  id
+      |)
+      |DO UPDATE SET
+      |  list = :list;
+      """.trimMargin(), tempFolder, dialectPreset = DialectPreset.SQLITE_3_24)
+
+    fun IntermediateType.compareTo(column: SqlColumnDef) {
+      assertThat(sqliteType).isEqualTo(IntermediateType.SqliteType.INTEGER)
+      assertThat(javaType).isEqualTo(column.typeName)
+      assertThat(name).isEqualTo(column.columnName)
+      assertThat(this.column).isSameAs(column)
+    }
+
+    val (idColumn, listColumn) = file.findChildrenOfType<SqlColumnDef>().toList()
+    file.findChildrenOfType<SqlBindExpr>().map { it.argumentType() }.forEach {
+      when (it.name) {
+        "id" -> {
+          assertThat(it.sqliteType).isEqualTo(IntermediateType.SqliteType.INTEGER)
```

Comment:
did you mean to use the helper function you made above in these two cases? `it.compareTo(idColumn)`

Reply:
Whoops forgot to take that out. I decided being explicit looked better 😁

Reply:
I'll remove it shortly

Reply:
Done

Reply:
forgot to push?

Reply:
It's been a long year :see_no_evil:

Reply:
i feel that